resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1225,7 +1231,8 @@ private[spark] class Client(
         if (log.isDebugEnabled) {
           logDebug(formatReportDetails(report, getDriverLogsLink(report)))
         } else if (lastState != state) {
-          logInfo(formatReportDetails(report, getDriverLogsLink(report)))
+          logInfo(log"${MDC(LogKey.REPORT_DETAILS,
```

Comment:
I am not sure if it is necessary to expand all variables in `ormatReportDetails(report, getDriverLogsLink(report))`, or is it acceptable in its current form?
https://github.com/apache/spark/blob/61dc9d991373c01d449a8ed26d9bfd7eb93f9301/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala#L1307-L1326

Reply:
It would be nice to expand this. But I am also fine to keep it as it is now.